aten/src/ATen/native/cuda/DistanceKernel.cu

```diff
@@ -79,19 +80,18 @@ struct dists {
 
 template <typename scalar_t, typename F>
 __global__ static void pdist_kernel_cuda_impl(scalar_t * result, const scalar_t * self, const int64_t n, const int64_t m, const scalar_t p) {
-  const int l = blockIdx.x;
-  const int k = blockIdx.y;
-  const int stride = blockDim.y;
+  const int k = blockIdx.x;
+  const int stride = blockDim.x;
 
-  float n2 = n - .5;
+  double n2 = n - .5;
   // The -1 accounts for floating point truncation issues
   int64_t i = static_cast<int64_t>((n2 - device_sqrt<scalar_t>(n2 * n2 - 2 * k - 1)));
```

Comment:
Without testing this, my hunch is that this should be `double` and not whatever precision is actually being used.

```suggestion
  int64_t i = static_cast<int64_t>((n2 - device_sqrt<double>(n2 * n2 - 2 * k - 1)));
```